### ISSUE TITLE
Request example shows wrong URI format

### ISSUE THREAD

**Issue by KentNordstrom** [microsoftgraph/microsoft-graph-docs#15311]
In example it says...
DELETE /groups/{id}/owners/{id}/$ref
but that's causing a (400) bad request

What works i using it without the /$ref in the end...
DELETE /groups/{id}/owners/{id}
the $ref is indead needed for the adding of owners (POST method), but don't work and is not needed when deleting owner.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2b4e8438-09f9-b27d-ceed-559d9f142211
* Version Independent ID: dc761c5d-1357-5fdb-f2e3-fd687ef89346
* Content: [Remove group owner - Microsoft Graph v1.0](https://docs.microsoft.com/en-us/graph/api/group-delete-owners?view=graph-rest-1.0&tabs=http&source=docs)
* Content Source: [api-reference/v1.0/api/group-delete-owners.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/main/api-reference/v1.0/api/group-delete-owners.md)
* Product: **groups**
* Technology: **microsoft-graph**
* GitHub Login: @Jordanndahl
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by FaithOmbongi** [comment]
Thank you @KentNordstrom for raising this issue. An ask, could you confirm that the user ID you deleted using that method is still in your tenant?

**Comment by KentNordstrom** [comment]
Yes, it's still present and I can use POST to add it back as Owner... and adding Owner has the correct docs... you need the $ref since the body contains a reference... but Remove does not have a body, so I guess the $ref indicating what type of information is in the body is not needed.

**Comment by FaithOmbongi** [comment]
Thanks for the confirmation @KentNordstrom. Unfortunately, I'm not able to replicate this in my tenant. The documented `DELETE /groups/{id}/owners/{id}/$ref` works as intended. When I try `DELETE /groups/{id}/owners/{id}` on the other hand, it also deletes the user from the tenant.

![image](https://user-images.githubusercontent.com/14026935/146966780-44e52eda-9b46-4bc8-bdbd-ed6b122b8c2c.png)

Let me escalate this to the product team to confirm. Could you share your correlation Id?

**Comment by KentNordstrom** [comment]
It looks like this is a bit worse than it first seemed... it depends on how many owners the group has :-(...

I have two groups... one where there are mulitple users listed as owner and one where my user is the only user listed as owner. The id 9c6acf28-340a-4915-8ab3-3577779f5d96 you see is a serviceprincipal also listed as owner for the groups.

First i try with the group with mulitple owners and DELETE works as docs say...
PS C:\> $GroupId = "6d148de3-111b-4287-a1ce-4efdd18e86e2" #Multiple User as Owner
PS C:\> $UserId = "e6c30262-b12f-4760-a020-c7321276ca2f"
PS C:\> $PropertiesToGet = "id,userPrincipalName"
PS C:\> $OwnersUri = "https://graph.microsoft.com/beta/groups/" + $GroupId + "/owners?`$select=$PropertiesToGet"
PS C:\> (Invoke-RestMethod -Method GET -Uri $OwnersUri -Headers $GraphHeader).value.id
97f7171e-3d26-4405-a71e-b43bae1aee7b
aa19963b-e10c-49f6-9bd3-104c5a77aaf7
9c6acf28-340a-4915-8ab3-3577779f5d96
e6c30262-b12f-4760-a020-c7321276ca2f

PS C:\> $RemoveOwnerUri = "https://graph.microsoft.com/v1.0/groups/" + $GroupId + "/owners/" + $UserId +"/`$ref"
PS C:\> Invoke-RestMethod -Method DELETE -Uri $RemoveOwnerUri -Headers $GraphHeader

Then I try the same thing with group where it's the only user...
PS C:\> $GroupId = "653178de-d79f-4e78-9ab7-44bc05969574" #Only one User as Owner
PS C:\> $OwnersUri = "https://graph.microsoft.com/beta/groups/" + $GroupId + "/owners?`$select=$PropertiesToGet"
PS C:\> (Invoke-RestMethod -Method GET -Uri $OwnersUri -Headers $GraphHeader).value.id
9c6acf28-340a-4915-8ab3-3577779f5d96
e6c30262-b12f-4760-a020-c7321276ca2f

PS C:\> $RemoveOwnerUri = "https://graph.microsoft.com/v1.0/groups/" + $GroupId + "/owners/" + $UserId +"/`$ref"
PS C:\> Invoke-RestMethod -Method DELETE -Uri $RemoveOwnerUri -Headers $GraphHeader
Invoke-RestMethod : The remote server returned an error: (400) Bad Request.
At line:1 char:1
+ Invoke-RestMethod -Method DELETE -Uri $RemoveOwnerUri -Headers $Graph ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
I found this as part of a bug report from an automation we have and I now see why the bug report said it failed when removing the "last" owner but worked for adding/removing owners if there where more than one.

**Comment by FaithOmbongi** [comment]
Thanks, @KentNordstrom for investigating further and providing logs. The product team is looking into this.
cc @psaffaie

**Comment by psaffaie** [comment]
@KentNordstrom at first glance this looks to be expected behavior. Can you confirm what group type these groups are? Default behavior for an M365 group is to enforce at least 1 group owner, where a service principal cannot be the only owner of a group. 
So, if there is 1 user object and 1 service principal object, the user object cannot be removed as an owner of that group because a service principal cannot be the only owner.

**Comment by KentNordstrom** [comment]
It is indeed a Unified group... but nothing prevents such a group to have only a ServicePrincipal as owner. I'm guessing it might be different if the group has Teams enabled on it or not. We created these UnifiedGroups with only the ServicePrincipal as the owner.

(Invoke-RestMethod -Method Get -Uri $GroupUri -Headers $GraphHeader) | Select id,groupTypes

id                                   groupTypes                  
--                                   ----------                  
653178de-d79f-4e78-9ab7-44bc05969574 {DynamicMembership, Unified}

If I look at the current owners I have the user as one owner at the moment...
(Invoke-RestMethod -Method GET -Uri $OwnersUri -Headers $GraphHeader).value.id
9c6acf28-340a-4915-8ab3-3577779f5d96
e6c30262-b12f-4760-a020-c7321276ca2f

If I then remove the Owner without the /$ref in the Uri
$RemoveOwnerUri = "https://graph.microsoft.com/v1.0/groups/" + $GroupId + "/owners/" + $UserId
Invoke-RestMethod -Method DELETE -Uri $RemoveOwnerUri -Headers $GraphHeader

I get no errors and teh group now have "only" the ServicePrincipal as Owner...
(Invoke-RestMethod -Method GET -Uri $OwnersUri -Headers $GraphHeader).value.id
9c6acf28-340a-4915-8ab3-3577779f5d96

I do however seem like the user is indeed deleted at this point...
(Invoke-RestMethod -Method Get -Uri $UserUri -Headers $GraphHeader).id
Invoke-RestMethod : The remote server returned an error: (404) Not Found.

Not sure why I did not see this in my first testings... maybe the user object gets restored in a while by Azure AD connect when the user object gets synced back from on-prem AD.

In my opinion this should not be possible... calling the "groups" endpoint I should not be able to delete or update a user, to do that I should need to call the "users" endpoint. Does this also mean that if I onlye have access to manage groups I inherit the permisson to delete users this way using the graph? Huge security flaw if that's the case! (I'm sorry I do not have time to test this now)

And anyhow... if the reason for the error in the first place is that I'm not allowed to remove the "last" owner... any other call like deleting the user should also fail for the same reason.

**Comment by FaithOmbongi** [comment]
Hi @KentNordstrom, to summarize the feedback from engineering:

Both of the following queries work as follows:

- DELETE https://graph.microsoft.com/v1.0/groups/{id}/owners/{id}/$ref - the documented method. Removes the group owner (but not if the last owner of microsoft.graph.user type).
- DELETE https://graph.microsoft.com/v1.0/groups/%7bid%7d/owners/{id} - Removes the group owner _and_ deletes them from the tenant

So the 2nd behavior where the group owner is also deleted is actually the intended behavior.

You shared concerns about the behavior of the 2nd request (now confirmed as intended). I've passed this along to Engineering. Unfortunately, this forum is only for documentation issues and issues such as what you've raised are best followed up through [Microsoft Graph on Q&A](https://docs.microsoft.com/en-us/answers/topics/microsoft-graph-groups.html). I recommend filing these concerns there and scope the discussion to this intended behavior.

Closing this issue here.
